src/test/java/space/obminyashka/items_exchange/controller/AdvertisementControllerTest.java

```diff
@@ -102,4 +107,123 @@ void findPaginatedAsThumbnails_shouldReturnProperQuantityOfAdvertisementsThumbna
                 .andExpect(status().isOk())
                .andExpect(jsonPath("$.length()").value(advertisementRepository.findAll().size()));
     }
+
+    @Test
+    @WithMockUser(username = "admin")
+    void updateAdvertisement_shouldReturn400WhenNotValidAdvertisementSize() throws Exception {
+        AdvertisementDto existDtoForUpdate = AdvertisementDtoCreatingUtil
+                .createExistAdvertisementDtoForUpdateWithNotValidSize();
+
+        MvcResult mvcResult = mockMvc.perform(put("/adv")
+                .content(asJsonString(existDtoForUpdate))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isBadRequest())
+                .andReturn();
+
+        assertTrue(mvcResult
+                .getResponse()
+                .getContentAsString()
+                .contains("must be between 1 and 50 symbols"));
+    }
+
+    @Test
+    @WithMockUser(username = "admin")
+    void updateAdvertisement_shouldReturn400WhenNotValidAdvertisementLocation() throws Exception {
+        AdvertisementDto existDtoForUpdate = AdvertisementDtoCreatingUtil
+                .createExistAdvertisementDtoForUpdateWithNotValidLocation();
+
+        String expectedResponseArea =
+                "area: " + existDtoForUpdate.getLocation().getArea() + " must be between 2 and 100 symbols";
```

Comment:
You could use MessageSourceUtil to get the message for further comparring.

Reply:
ok, will be fixed